### PR TITLE
Better information and tools for setting root dir

### LINKED ISSUE #438: Default root dir not set on NERSC interactive queue nodes
Here's what I see on an interactive queue node at NERSC:
```
In [1]: import GCRCatalogs                                                      
/global/u1/j/jchiang8/dev/gcr-catalogs/GCRCatalogs/register.py:109: UserWarning: Default root dir has not been set!
  warnings.warn("Default root dir has not been set!")

In [2]: import socket                                                           

In [3]: socket.getfqdn()                                                        
Out[3]: 'nid00014'
```
This could be addressed by adding an `nid` entry in [site_rootdir.yaml](https://github.com/LSSTDESC/gcr-catalogs/blob/master/GCRCatalogs/site_config/site_rootdir.yaml) similar to the `nersc` entry.

### PR DESCRIPTION
**Pull request by JoanneBogart**
Fix #438

### CODE REVIEW

File: GCRCatalogs/register.py [hunk 1]
```diff
@@ -91,22 +91,39 @@ class RootDirManager:
     )
     _DICT_LIST_KEYS = ("catalogs",)
     _DESC_SITE_ENV = "DESC_GCR_SITE"
+    _NO_DEFAULT_ROOT_WARN = """
+       Default root dir has not been set; catalogs may not be found.
+       To set root dir by site 
+            $ export {}='sitename'
+       or, from within Python,
```
Comment (yymao) [inline GCRCatalogs/register.py:98]:
```suggestion
       or, from within Python and after you import GCRCatalogs,
```

File: GCRCatalogs/register.py [hunk 2]
```diff
@@ -91,22 +91,39 @@ class RootDirManager:
...
             for k, v in site_config.items():
                 if k in self._site_info:
                     self._default_root_dir = v
                     break
 
         if not self._default_root_dir:
-            warnings.warn("Default root dir has not been set!")
+            site_string = ' '.join(self._site_list)
+            warnings.warn(self._NO_DEFAULT_ROOT_WARN.format(self._DESC_SITE_ENV, site_string))
+            #warnings.warn("Default root dir has not been set!")
```
Comment (yymao) [inline GCRCatalogs/register.py:126]:
```suggestion
```

File: GCRCatalogs/register.py [hunk 3]
```diff
@@ -142,6 +159,17 @@ def root_dir(self, path):
             warnings.warn("root_dir has been set to '{}' but errors may occur when you try to access it: {}".format(path, e))
         self._custom_root_dir = path
 
+    def set_root_dir_by_site(self, site):
+        """
+        If *site* is a recognized site, set root_dir to corresponding value
+        """
+        if site in self._site_list:
+            self.root_dir = self._site_config[site]
+        else:
```
Comment (yymao) [inline GCRCatalogs/register.py:168]:
```suggestion
        try:
            self.root_dir = self._site_config[site]
        except KeyError:
```

File: GCRCatalogs/register.py [hunk 4]
```diff
@@ -142,6 +159,17 @@ def root_dir(self, path):
             warnings.warn("root_dir has been set to '{}' but errors may occur when you try to access it: {}".format(path, e))
         self._custom_root_dir = path
 
+    def set_root_dir_by_site(self, site):
```
Comment (yymao) [inline GCRCatalogs/register.py:162]: Make `site_list` an on-the-fly property
```suggestion
    @property
    def site_list(self):
        return list(self._site_config)

    def set_root_dir_by_site(self, site):
```

File: GCRCatalogs/register.py [hunk 5]
```diff
@@ -142,6 +159,17 @@ def root_dir(self, path):
...
+        """
+        If *site* is a recognized site, set root_dir to corresponding value
+        """
+        if site in self._site_list:
+            self.root_dir = self._site_config[site]
+        else:
+            site_string = ' '.join(_config_register._site_list)
```
Comment (yymao) [inline GCRCatalogs/register.py:169]: Use the new public `site_list` property
```suggestion
            site_string = ' '.join(_config_register.site_list)
```

File: GCRCatalogs/register.py [hunk 6]
```diff
@@ -512,6 +540,17 @@ def set_root_dir(path):
...
+    """
+    Sets runtime root_dir to path corresponding to *site*.
+    """
+    _config_register.set_root_dir_by_site(site)
+
+def get_site_list():
+    """
+    Return list of recognized sites
+    """
+    return _config_register._site_list
```
Comment (yymao) [inline GCRCatalogs/register.py:553]: Use the public `site_list` property
```suggestion
    return _config_register.site_list
```

File: GCRCatalogs/register.py [hunk 7]
```diff
@@ -91,22 +91,39 @@ class RootDirManager:
...
         self._custom_root_dir = None
+        self._site_list = []
+        self._site_config = {}
         self._site_info = self._get_site_info()
 
         if self._site_config_path:
             site_config = load_yaml_local(self._site_config_path)
+            self._site_config = site_config
+            self._site_list = list(site_config.keys())
             for k, v in site_config.items():
```
Comment (yymao) [inline GCRCatalogs/register.py:118]:
```suggestion
            self._site_config = load_yaml_local(self._site_config_path)
            for k, v in self._site_config.items():
```

File: GCRCatalogs/register.py [hunk 8]
```diff
@@ -91,22 +91,39 @@ class RootDirManager:
...
                 if k in self._site_info:
                     self._default_root_dir = v
                     break
 
         if not self._default_root_dir:
-            warnings.warn("Default root dir has not been set!")
+            site_string = ' '.join(self._site_list)
```
Comment (yymao) [inline GCRCatalogs/register.py:124]: Use public `site_list`
```suggestion
            site_string = ' '.join(self.site_list)
```

File: GCRCatalogs/register.py [hunk 9]
```diff
@@ -91,22 +91,39 @@ class RootDirManager:
...
+
+       If standard site values for root dir are not appropriate, use
+            GCRCatalogs.set_root_dir('my_root_dir')
+    """
 
     def __init__(self, site_config_path=None):
         self._site_config_path = site_config_path
         self._default_root_dir = None
         self._custom_root_dir = None
+        self._site_list = []
```
Comment (yymao) [inline GCRCatalogs/register.py:110]:
```suggestion
```

File: GCRCatalogs/register.py [hunk 10]
```diff
@@ -91,22 +91,39 @@ class RootDirManager:
     )
     _DICT_LIST_KEYS = ("catalogs",)
     _DESC_SITE_ENV = "DESC_GCR_SITE"
+    _NO_DEFAULT_ROOT_WARN = """
+       Default root dir has not been set; catalogs may not be found.
+       To set root dir by site
```
Comment (yymao) [inline GCRCatalogs/register.py:96]:
```suggestion
       You can specify the site as an environment variable before you import GCRCatalogs,
```

File: GCRCatalogs/register.py [hunk 11]
```diff
@@ -91,22 +91,39 @@ class RootDirManager:
...
+    _NO_DEFAULT_ROOT_WARN = """
+       Default root dir has not been set; catalogs may not be found.
+       To set root dir by site 
+            $ export {}='sitename'
+       or, from within Python,
+            GCRCatalogs.set_root_dir_by_site('sitename')
+       where sitename is one of ({})
+
+       If standard site values for root dir are not appropriate, use
+            GCRCatalogs.set_root_dir('my_root_dir')
```
Comment (yymao) [inline GCRCatalogs/register.py:103]:
```suggestion
       If you want to use a non-standard root dir, use
            GCRCatalogs.set_root_dir('/path/to/your/root_dir')
```

**Comment by JoanneBogart** [comment]
I agree that site_list should not be cached separately from site_config, even though it's unlikely either would be changed.
I have no objection to any of the other suggestions, either, so I've committed them all.

File: GCRCatalogs/register.py [hunk 12]
```diff
@@ -142,6 +155,21 @@ def root_dir(self, path):
...
+    def set_root_dir_by_site(self, site):
+        """
+        If *site* is a recognized site, set root_dir to corresponding value
+        """
+        try:
+            self.root_dir = self._site_config[site]
+        except KeyError:
+            site_string = ' '.join(_config_register.site_list)
+            warnings.warn(f"Unknown site '{site}'.\nAvailable sites are: {site_string}\nroot_dir is unchanged")
+
```
Comment (yymao) [inline GCRCatalogs/register.py:171]: Removing an extra empty line
```suggestion
```

**Pull request merged**